src/client/datascience/notebook/notebookControllerManager.ts

```diff
@@ -391,7 +391,7 @@ export class NotebookControllerManager implements INotebookControllerManager, IE
 
         // Prep so that we can track the selected controller for this document
         traceInfoIfCI(`Clear controller mapping for ${getDisplayPath(document.uri)}`);
-        const loadControllersPromise = this.loadNotebookControllers();
+        void this.loadNotebookControllers();
```

Comment:
Its ok if the controllers haven't loaded, 
We can create the controllers as required.
After all, that's what we do, we get controller information and create the controller.
If we call that again its not re-created.

This way, we don't have to wait for all 10-20 or 50 controllers to get loaded. 
I.e. we can create controllers one by one based on the information available.

Result, we can now run button will be available even earlier.

Got more changes that will imporve this scenario & make this change all the more useful (separate PR).